packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -30,6 +30,8 @@ class SwipeableDrawer extends React.Component {
 
   isSwiping = null;
 
+  paperRef = null;
```

Comment:
We only keep track of the class variables that need a default value. `isSwiping = null` has a strong implication in the logic.

```suggestion
```

Reply:
What's the difference between `undefined` and `null` for `isSwiping`?

Reply:
Ok, my bad. Now, we are moving away from class components, do as you like, it will soon not matter :).

Reply:
My point is that I prefer some sort of declaration. 

You can do this implicit with types. 

Plain JS allows some very bad patterns IMO. Reading the code it can be very surprising to find a instance method/var that has only been declared later in the code. These things should be done at a single place which helps understanding the code. This is not an issue for the original author but everybody reading the code for the first time will go "huh? where does this come from?" and then you have to search the whole code instead of simply reading the declaration list in the constructor.

I get that this has runtime implications so I'm totally fine with removing it. But we need some place where this is documented. Can be done with types or JSDoc which comes with a host of other problems (not verified, will become outdated etc). So I'd rather at a few bytes to improve code readability. You could always write a compiler that strips `undefined` initializers.

> Now, we are moving away from class components, do as you like, it will soon not matter :).

Also an important part. With function components we need explicit initializers anyway.

Reply:
https://github.com/mui-org/material-ui/pull/13022

Reply:
> The motivation is in #13022.

> I get that this has runtime implications so I'm totally fine with removing it.

Code is not always written the way it is to implement a specific behavior. You shouldn't write code so that a computer can understand it but so that the person after you (or future self) understands it. 

We could've easily written 
```diff
-someInstanceField = null;
+someInstanceField;
```

and add a babel plugin that strips those. The [typescript compiler is already doing this](https://www.typescriptlang.org/play/index.html#src=class%20Initializers%20%7B%0D%0A%20%20field%3B%0D%0A%7D).

Reply:
Definitely, the problem could have been better handled. It was a tradeoff.